app/main/views/frameworks.py

```diff
@@ -700,56 +719,49 @@ def signature_upload(framework_slug):
                 disposition_type='inline'  # Embeddeding PDFs in admin pages requires 'inline' and not 'attachment'
             )
 
-            agreement_id = supplier_framework.get("agreementId")
             data_api_client.update_framework_agreement(agreement_id, {"signedAgreementPath": upload_path},
                                                        current_user.email_address)
 
             session['signature_page'] = request.files['signature_page'].filename
 
-            data_api_client.create_audit_event(
-                audit_type=AuditTypes.upload_signed_agreement,
-                user=current_user.email_address,
-                object_type="suppliers",
-                object_id=current_user.supplier_id,
-                data={
-                    "upload_signed_agreement": request.files['signature_page'].filename,
-                    "upload_path": upload_path
-                })
-
-            return redirect(url_for(".contract_review", framework_slug=framework_slug))
+            return redirect(url_for(".contract_review", framework_slug=framework_slug, agreement_id=agreement_id))
 
     return render_template(
         "frameworks/signature_upload.html",
+        agreement=agreement,
         framework=framework,
         signature_page=signature_page,
         upload_error=upload_error,
     ), 400 if upload_error else 200
 
 
-@main.route('/frameworks/<framework_slug>/contract-review', methods=['GET', 'POST'])
+@main.route('/frameworks/<framework_slug>/<int:agreement_id>/contract-review', methods=['GET', 'POST'])
 @login_required
-def contract_review(framework_slug):
-    framework = get_framework(data_api_client, framework_slug)
+def contract_review(framework_slug, agreement_id):
+    framework = get_framework(data_api_client, framework_slug, allowed_statuses=['standstill', 'live'])
     supplier_framework = return_supplier_framework_info_if_on_framework_or_abort(data_api_client, framework_slug)
-    agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
-    signature_page = agreements_bucket.get_key(supplier_framework['agreementPath'])
+    agreement = data_api_client.get_framework_agreement(agreement_id)['agreement']
+    check_agreement_is_related_to_supplier_framework_or_abort(agreement, supplier_framework)
 
-    # if supplier_framework doesn't have a name or a role or the agreement file, then 404
+    # if framework agreement doesn't have a name or a role or the agreement file, then 404
     if not (
-        supplier_framework['agreementDetails'] and
-        supplier_framework['agreementDetails'].get('signerName') and
-        supplier_framework['agreementDetails'].get('signerRole') and
-        supplier_framework['agreementPath']
+        agreement.get('signedAgreementDetails') and
+        agreement['signedAgreementDetails'].get('signerName') and
+        agreement['signedAgreementDetails'].get('signerRole') and
+        agreement.get('signedAgreementPath')
```

Comment:
This doesn't work if the `signedAgreementDetails` key exists with a `None` value though, (e.g. `{'signedAgreementDetails': None, 'signedAgreementPath': None, 'frameworkSlug': 'g-cloud-8', 'id': 234, 'supplierId': 2345}`) so I prefer the version as it.